### PR TITLE
Make it easier to mock GPS positions as though the device was moving …

### PR DESCRIPTION
**Pull request by Crashthatch**
…around a map (eg. provide GPS position of 200m from starting point in x direction).

----

Combine functionality of TransformLocationProvider into EditorLocationProvider so don't have to keep switching which one is being used while testing.

### CODE REVIEW

File: sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs [hunk 1]
```diff
@@ -21,23 +22,38 @@ public class EditorLocationProvider : AbstractEditorLocationProvider
...
 		/// </summary>
 		[SerializeField]
-		[Range(0, 359)]
-		float _heading;
+		Transform _targetTransform;
+
+		[SerializeField]
+		private AbstractMap _map;
+
+		private bool mapInitialized = false;
```
Comment (david-rhodes) [inline sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs]: Where is this ever changed to true?

Comment (david-rhodes) [reply sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs]: Edit: I know what it's for, but it's never set accordingly.

Comment (Crashthatch) [reply sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs]: You're right. I selectively committed hunks to create this PR, so it is set in my version, but not in this commit.
If you're going to create a new PR anyway can I leave it to you to fix?

Comment (david-rhodes) [reply sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs]: No problem!

**Pull request closed without merging**